src/cargo/util/toml/mod.rs

```diff
@@ -1095,28 +1093,187 @@ pub struct TomlWorkspace {
     resolver: Option<String>,
 
     // Properties that can be inherited by members.
+    package: Option<InheritableFields>,
+    dependencies: Option<BTreeMap<String, TomlDependency>>,
+
+    // Note that this field must come last due to the way toml serialization
+    // works which requires tables to be emitted after all values.
+    metadata: Option<toml::Value>,
+}
+
+/// A group of fields that are inheritable by members of the workspace
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct InheritableFields {
```

Comment:
Note for the future: this is a case where it helps if you split this up into multiple commits, one for the move and then one for the actual structure change.